### PR TITLE
Feature/itl 4

### PR DESCRIPTION
**Pull request by doramagyari**

### CODE REVIEW

File: src/components/ui/BaseCard.vue [hunk 1]
```diff
@@ -128,4 +132,9 @@ function formatDate(date: Date): string {
 function toggleDropDown(dropDownClicked: boolean) {
   isDropDownClicked.value = dropDownClicked;
 }
+
+function todoChecked() {
```
Comment (kincsokiss) [inline src/components/ui/BaseCard.vue]: Please make sure to use the proper function name and use actions or use `on` prefix if it's triggered for a specific event. For example: `checkTodo` or `onTodoChecked`

File: src/components/ui/BaseCard.vue [hunk 2]
```diff
@@ -128,4 +132,9 @@ function formatDate(date: Date): string {
...
+  localTodo.value.complete = !localTodo.value.complete;
+  emit("todo-checked", localTodo.value.id, localTodo.value.complete);
```
Comment (kincsokiss) [inline src/components/ui/BaseCard.vue]: Same in the emit: `on-todo-checked`

File: src/components/ui/BaseCheck.vue [hunk 3]
```diff
@@ -1,17 +1,15 @@
...
+  </div>
 </template>
 
 <script setup lang="ts">
 import { ref } from "vue";
 import ellipseSvg from "../../assets/svg/ellipse.svg";
 import ellipseCheckedSvg from "../../assets/svg/ellipse-checked.svg";
 
-const emit = defineEmits(["update:modelValue"]);
+const emit = defineEmits(["update:modelValue", "checked-todo"]);
```
Comment (kincsokiss) [inline src/components/ui/BaseCheck.vue]: If you created a general component for Check, you shouldn't add the `todo` here in order not to change this to a specific component for Todos. 
In this case you can add everything as general as you can.
 For example:
 Change the `checked-todo` to `on-checked`
 The action should be handled here which is the checked, you don't need to specify what is checked.

**Review by kincsokiss — CHANGES_REQUESTED** [review]

File: src/components/pages/TodoHome.vue [hunk 4]
```diff
@@ -49,4 +50,17 @@ function handleToggleButtons(id: string) {
 function isClicked(id: string) {
   return currentClickedId.value === id;
 }
+
+function moveChecked(id: string, isChecked: boolean) {
+  const index = todos.value.findIndex((todo) => todo.id === id);
+  if (index !== -1) {
```
Comment (gerebdavid) [inline src/components/pages/TodoHome.vue]: in cases like this its better to reverse the condition and use early return, so your code wont be indented a lot.

File: src/components/ui/BaseCard.vue [hunk 5]
```diff
@@ -3,49 +3,54 @@
     class="cursor-pointer box-border bg-white border-2 border-black rounded-lg sm:mx-4 sm:mb-8 lg:mb-12 p-1"
     @click="editCard"
   >
-    <div class="flex justify-between mt-3">
+    <div class="flex justify-between mt-2">
       <BaseInput
+        class="sm:text-xl lg:text-2xl"
         v-model="localTodo.title"
```
Comment (gerebdavid) [inline src/components/ui/BaseCard.vue:9]: v-model should be higher than class

File: src/components/ui/BaseCard.vue [hunk 6]
```diff
@@ -3,49 +3,54 @@
...
         :class="{ 'opacity-30': isDropDownClicked }"
       ></BaseInput>
       <BaseDropdown
+        class="mt-0 sm:text-sm lg:text-lg"
+        :todo="todo"
         v-model="localTodo.priority"
```
Comment (gerebdavid) [inline src/components/ui/BaseCard.vue:15]: v-model should be higher than class

File: src/components/ui/BaseCard.vue [hunk 7]
```diff
@@ -3,49 +3,54 @@
...
       <BaseTextarea
+        class="sm:text-sm lg:text-lg"
         v-model="localTodo.description"
         :class="{ 'opacity-30': isDropDownClicked }"
       ></BaseTextarea>
       <BaseCheck
         v-show="!isClicked"
         v-model="localTodo.complete"
         @click.stop="saveChanges"
+        @on-checked="onTodoChecked"
```
Comment (gerebdavid) [inline src/components/ui/BaseCard.vue:38]: do you need a click event here?

File: src/components/ui/BaseCard.vue [hunk 8]
```diff
@@ -128,4 +135,9 @@ function formatDate(date: Date): string {
 function toggleDropDown(dropDownClicked: boolean) {
   isDropDownClicked.value = dropDownClicked;
 }
+
+function onTodoChecked() {
```
Comment (gerebdavid) [inline src/components/ui/BaseCard.vue:139]: why do you need a separate emit for todo checked? cant you use update-todo here?

File: src/components/ui/BaseCard.vue [hunk 9]
```diff
@@ -3,49 +3,54 @@
...
     </div>
     <div class="flex justify-between">
       <BaseTextarea
+        class="sm:text-sm lg:text-lg"
         v-model="localTodo.description"
         :class="{ 'opacity-30': isDropDownClicked }"
       ></BaseTextarea>
       <BaseCheck
         v-show="!isClicked"
         v-model="localTodo.complete"
```
Comment (gerebdavid) [inline src/components/ui/BaseCard.vue:36]: Either have v-model or on-checked emit, using both makes no sense

File: src/components/ui/BaseCheck.vue [hunk 10]
```diff
@@ -1,17 +1,19 @@
 <template>
-  <img
-    :src="currentSvg"
-    class="sm:hidden lg:block lg:pr-4"
-    @click="clickedSvg"
-  />
+  <div @click.stop="clickedSvg">
```
Comment (gerebdavid) [inline src/components/ui/BaseCheck.vue:2]: why do you need a click event on the div and another click event on the image?

**Pull request merged**